plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -3968,4 +3917,13 @@ private static boolean isQueryPartitionFilterRequiredForTable(ConnectorSession s
         return isQueryPartitionFilterRequired(session) &&
                 requiredSchemas.isEmpty() || requiredSchemas.contains(schemaTableName.getSchemaName());
     }
+
+    private static CatalogSchemaTableName appendSuffixToTableName(CatalogSchemaTableName name, String suffix)
+    {
+        return new CatalogSchemaTableName(
+                name.getCatalogName(),
+                new SchemaTableName(
+                        name.getSchemaTableName().getSchemaName(),
+                        name.getSchemaTableName().getTableName() + suffix));
+    }
```

Comment:
since we've simplified the logic enough that we only do this in one place, it seems we can revert the change to extract this to a separate method, WDYT?

Reply:
I agree, reverted this change and made it inline.